pkgs/applications/misc/citations/default.nix

```diff
@@ -0,0 +1,77 @@
+{
+  citations,
+  darwin,
+  desktop-file-utils,
+  fetchFromGitLab,
+  gettext,
+  glib,
+  gtk4,
+  gtksourceview5,
+  lib,
+  libadwaita,
+  meson,
+  ninja,
+  pkg-config,
+  poppler,
+  rustPlatform,
+  stdenv,
+  testers,
+  wrapGAppsHook4,
+}:
+stdenv.mkDerivation rec {
+  pname = "citations";
+  version = "0.5.1";
+
+  src = fetchFromGitLab {
+    domain = "gitlab.gnome.org";
+    owner = "World";
+    repo = pname;
+    rev = version;
+    hash = "sha256-QPK6Nw0tDdttUDFKMgThTYMTxGXsn5OReqf1LNAai7g=";
+  };
+
+  cargoDeps = rustPlatform.fetchCargoTarball {
+    inherit src;
+    name = "${pname}-${version}";
+    hash = "sha256-Kounxi4JxoU4+rWMWNB8rzTyG3MDKYD0OzYfAHwm6bY=";
+  };
+
+  nativeBuildInputs = [
+    desktop-file-utils
+    gettext
+    glib
+    meson
+    ninja
+    pkg-config
+    rustPlatform.cargoSetupHook
+    rustPlatform.rust.cargo
+    rustPlatform.rust.rustc
+    wrapGAppsHook4
+  ];
+
+  buildInputs = [
+    glib
+    gtk4
+    gtksourceview5
+    libadwaita
+    poppler
+  ]
+  ++ lib.optional stdenv.isDarwin [
+    darwin.apple_sdk.frameworks.Foundation
+  ];
+
+  doCheck = true;
+
+  passthru.tests.version = testers.testVersion {
+    package = citations;
```

Comment:
It might be nicer to use the [`finalAttrs.finalPackage` expression](https://nixos.org/manual/nixpkgs/stable/#mkderivation-recursive-attributes) so that this works even when overridden.